site/content/contribute/gsoc/_index.md

```diff
@@ -0,0 +1,223 @@
+3. Send push and desktop notification summaries when users come out of Do Not Disturb: When a user goes back online after being in Do Not Disturb mode, this feature would send a digest push and desktop notification (if enabled) like "You missed X Direct Messages and 5 channel mentions while in Do Not Disturb".
+
+-   **Expected outcomes**: Implementation of the backend and frontend features for at least two out of the three tasks listed.
+-   **Skills required**: React, Go
+-   **Difficulty**: Medium
+-   **Mentors**: Alejandro García Montoro, Eric Sethna
+
+### Control link previews in posts
+
+When a message contains a URL, a preview of the linked content is shown in the post. There are situations when a plugin or an app may post several messages, each of them with a different link, which causes the channel where the messages are posted to be quickly filled with the previews. This project tries to improve this poor experience with two different tasks:
+
+1.  Modify the endpoint [`POST /posts`](https://api.mattermost.com/#tag/posts/paths/~1posts/post) in the REST API to add a property that would let the submitter control whether the posted message will contain a preview of the link in the post. This information needs to be properly communicated to the client for the preview to be enabled or disabled.
+2.  Implement a new feature, probably modifying the [`PUT /posts`](https://api.mattermost.com/#tag/posts/paths/~1posts~1{post_id}/put) endpoint, to dynamically enable and disable the preview of the links in specific, existing posts.
+
+This project is open to ideas that should be reflected in the student proposal.
+
+-   **Expected outcomes**: Design of the feature, modification of the endpoint to implement it and implementation of the dynamic toggling of the preview, both in the backend and the frontend.
+-   **Skills required**: React, Go
+-   **Difficulty**: Easy
+-   **Mentors**: Daniel Espino García, Alejandro García Montoro
+
+### Add the concept of _Importance_ to messages
+
+The importance of a message varies greatly, depending on its content, the sender, the receiver or even the channel that it is being shared on. However, there is no functional difference among all these messages, and all of them are posts with the same importance for the application. This can be a problem in several situations: Think, for example, of users across different time zones or users that are on vacation; when these users are back online, they can be easily overwhelmed with the number of unread messages and notifications. There are also situations where you want to send a low priority message that does not need the immediate attention of the recipient. And others where you need a quick response or, at least, an acknowledgment.
```

Comment:
```suggestion
The importance of a message is subjective and varies greatly, depending on the content, the sender, the receiver, or even the channel where it is being shared. However, there is no functional difference between such messages, and the application regards them as having equal importance today. This can be a problem in several situations. For example, image users in different time zones or users that are on vacation -- when these users are back online, they could be easily overwhelmed with the number of unread messages and notifications. There are also situations where you want to send a low priority message that does not need the immediate attention of the recipient. And others where you need a quick response or, at least, an acknowledgment.
```

Reply:
Is "image users" intended here?

Reply:
Great catch! I've updated my suggestion.